paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +130,74 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
```

Comment:
I'm concerned that this won't really reflect what authors really want. Think about yelp-main with potentially 1k instances across multiple clusters.

I think we are going to have to separate "what we deploy to" (deploy group) from "what we wait on" (clusters/instances?)

"Of course" people probably don't want to wait on batch jobs, do they only want to wait on the "main" stuff? Do they care about their deploy in testopia? (even though their deploy group is "everywhere")